landsites_tools/interface_settings.py

```diff
@@ -23,91 +28,296 @@ class SettingsParser:
     """NorESM Land Sites Platform settings parser"""
 
     def __init__(self, path_settings):
-        """Parse the NorESM Land Site Platform settings file
+        """
+        Parse the NorESM Land Site Platform settings file
         Arguments:
             paths_settings: path to settings file
         Returns:
             instance of SettingsParser class
         """
-        # Settings file path
+        ### Load user settings into a ConfigParser instance (low-level interface)
+        self.interface = ConfigParser(interpolation=ExtendedInterpolation(),
+                                      inline_comment_prefixes='#')
+        # Path to settings file
         self.path_settings = Path(path_settings).absolute()
-        # Default information
+        self.interface.read(self.path_settings)
+
+        ### Default information
+        self.version = landsites_tools.__version__
         self.dir_platform = Path(__file__).absolute().parents[1]
-        self.dir_info = self.dir_platform / 'data'
+        self.dir_info = self.dir_platform / 'data' / '.nlp'
         self.path_sites_table = self.dir_info / 'sites.json'
         self.info_all_sites = pd.read_json(self.path_sites_table,
                                            orient='index')
         self.path_constraints = self.dir_info / 'settings_constraints.json'
-        self.constraints = json.load(open(self.path_constraints, 'r'))
-        # Load user settings into a ConfigParser instance (low-level interface)
-        self.interface = ConfigParser(interpolation=ExtendedInterpolation(),
-                                      inline_comment_prefixes='#')
-        self.interface.read(self.path_settings)
-        # Parse user settings to convenient types
-        self.dir_cases = self.get_path('path', 'dir_cases')
-        self.dir_input = self.get_path('path', 'dir_input')
-        self.dir_output = self.get_path('path', 'dir_output')
+
+        ### Read parameter dictionary
+        with open(self.path_constraints, 'r') as constraints_file:
+            self.constraints = json.load(constraints_file)
+
+        ### Parse user settings to convenient types
+        # Paths
+        self.dir_cases = self.dir_platform / self.get_path('path', 'dir_cases',
+         'dir')
+        self.dir_input = self.dir_platform / self.get_path('path', 'dir_input',
+         'dir')
+        self.dir_output = self.dir_platform / self.get_path('path',
+        'dir_output', 'dir')
```

Comment:
In my opinion, keeping the cases, input, and output in dedicated folders within the project should _prevent_ confusion as it will be easier to document what exactly the standard target folders mean and do. Plus, people who are not used to working with directory structures as long as there is no GUI (based on teaching experiences) would not need to worry about this part. The newest update allows providing no target dirs (--> std folders with .gitkeep files), relative to the project root, and absolute to some other place. Happy to discuss, though!

`is_valid_path()` is now a single function in the utils folder.